src/taker/api.rs

```diff
@@ -976,14 +971,12 @@ impl Taker {
                 next_peer_hashlock_pubkeys,
                 next_peer_hashlock_keys_or_nonces,
             ) = if self.ongoing_swap_state.taker_position == TakerPosition::LastPeer {
-                let (my_recv_ms_pubkeys, my_recv_ms_nonce): (Vec<_>, Vec<_>) =
-                    (0..self.ongoing_swap_state.swap_params.tx_count)
-                        .map(|_| generate_keypair())
-                        .unzip();
-                let (my_recv_hashlock_pubkeys, my_recv_hashlock_nonce): (Vec<_>, Vec<_>) = (0
-                    ..self.ongoing_swap_state.swap_params.tx_count)
-                    .map(|_| generate_keypair())
-                    .unzip();
+                let (my_recv_ms_pubkey, my_recv_ms_nonce) = generate_keypair();
+                let (my_recv_hashlock_pubkey, my_recv_hashlock_nonce) = generate_keypair();
+                let my_recv_ms_pubkeys = vec![my_recv_ms_pubkey];
+                let my_recv_ms_nonce = vec![my_recv_ms_nonce];
+                let my_recv_hashlock_pubkeys = vec![my_recv_hashlock_pubkey];
+                let my_recv_hashlock_nonce = vec![my_recv_hashlock_nonce];
```

Comment:
Just add the vecs in the return statement, instead of having another variable. cleaner code.

They should not be vecs anymore, but changing that would require too much refactoring, so its fine for now.